### ISSUE TITLE
[SUGGESTION] FFMpeg's noise reduction

### ISSUE THREAD

**Issue by MarcoRavich** [MattMcManis/Axiom#73]
I recently encoded an old documentary thas has a "disturbing" grain, so I believe that would be great for encoder GUI to let users exploit denoise algorithms available in FFMpeg (as HandBrake already does):

- http://ffmpeg.org/ffmpeg-filters.html#nlmeans-1
- http://ffmpeg.org/ffmpeg-filters.html#atadenoise
- http://ffmpeg.org/ffmpeg-filters.html#hqdn3d-1
- http://ffmpeg.org/ffmpeg-filters.html#vaguedenoiser

Hope that inspires.

**Comment by MattMcManis** [comment]
I will work on adding this, thanks.

**Comment by MattMcManis** [comment]
Example commands of each noise filter:

```
-vf nlmeans=h=10:range=5:temporal=3

-vf atadenoise=s=7:p=7:0a=0:0b=0:1a=0:1b=0:2a=0:2b=0

-vf hqdn3d=0:0:8:8

-vf vaguedenoiser=2:2:6:85:15
```

For now I might have to do simple presets for each, `Heavy`, `Medium`, `Light`, as it will be difficult and take some time to add the parameter controls for each value, such as `threshold`, `luma`, `chroma`, `planes`.

**Comment by forart** [comment]
Yes, as Handbrake, you can simply put just some "presets" (eg. min-low-mid-high-max) per filter...

**Comment by MarcoRavich** [comment]
Yes, almost all GUIs uses presets...

**Comment by MattMcManis** [comment]
I've added the noise filters `nlmeans`, `hqdn3d`, `vaguedenoiser`, `removegrain` with `light`, `medium`, `strong` presets.  
Let me know how they work out.

---

I did not add `atadenoise` yet, if you can find any good presets for `light`, `medium`, `strong`, post them.

I found the other presets here:  
`nlmeans`
https://forum.handbrake.fr/viewtopic.php?f=4&t=30135&p=139077
`hqdn3d`
https://mattgadient.com/in-depth-look-at-de-noising-in-handbrake-with-imagevideo-examples/

---

I also added the `deblock` filter with `weak` and `strong`.  
https://ffmpeg.org/ffmpeg-filters.html#deblock

---

Press the update arrow in Axiom or download new version [here](https://github.com/MattMcManis/Axiom/releases).

**Comment by MarcoRavich** [comment]
Someone suggested to implement accellerated nlmeans_opencl @ FastFlix, wich sounds a really good idea:

https://github.com/cdgriffith/FastFlix/issues/202

Hope that inspires !
